collectors/monitoring_collector.go

```diff
@@ -171,6 +171,11 @@ func (c *MonitoringCollector) reportMonitoringMetrics(ch chan<- prometheus.Metri
 					if page == nil {
 						break
 					}
+					select {
+					case <-errChannel:
```

Comment:
It seems odd that you would swallow the error without printing anything. I am not sure how this should really work.